### PR TITLE
style: use prettier to enforce pedantic formatting on PR

### PR DESCRIPTION
**Pull request by philihp**
* Adds a devDependency prettier, which will enforce consistent styling.
* Usually I also add eslint, husky, and lint-staged, although probably overkill for just this lib.
* Additionally, this removes `package-lock.json`, [which shouldn't be committed](https://dev.to/gajus/stop-using-package-lock-json-or-yarn-lock-3ddi) for libraries when they are dependents of other projects.

### CODE REVIEW

File: typescript-client-sdk/.prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "semi": true,
+  "trailingComma": "none",
+  "singleQuote": false,
+  "printWidth": 123
```
Comment (hpx7) [inline typescript-client-sdk/.prettierrc.json]:
```suggestion
  "printWidth": 120
```

Comment (philihp) [reply typescript-client-sdk/.prettierrc.json]: Yeah we can go with 120. 123 was what caused the least amount of changes to the file as it is.

Comment (philihp) [reply typescript-client-sdk/.prettierrc.json]: Fixed. But also lol those were my changes anyway. Never too soon to add Prettier 😂

**Review by hpx7 — APPROVED** [review]
thanks!

**Pull request merged**